### PR TITLE
[W5.7][W13-4] Andy Chan Wai Yee

### PR DESCRIPTION
**Pull request by andlimey**

### CODE REVIEW

**Comment by lywjoel** [comment]
The sort command looks fantastic! Good job :)

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -72,6 +72,11 @@ Examples:
 Shows a list of all persons, along with their non-private details, in the address book. +
 Format: `list`
 
+== Listing all persons in sorted order : `sort`
+
+Shows a sorted list of all persons in lexicographical order, along with their non-private details, in the address book. +
+Format: `sort`
```
Comment (parvathy-sudhir-pillai) [inline docs/UserGuide.adoc:78]: Good job updating the user doc

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.Comparator;
+import java.util.List;
+
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:10]: Avoid stray line breaks by checking diff before submitting PR

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @Override
+    public CommandResult execute() throws UniquePersonList.DuplicatePersonException {
+        UniquePersonList allPersons = addressBook.getAllPersons();
+        List<Person> allPersonsList = allPersons.getInternalList();
+
+        //Sorts the Person List based on the names
+        allPersonsList.sort(Comparator.comparing(o -> o.getName().toString()));
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/SortCommand.java:30]: Can consider enhancing the sort command to sort based on particulars other than name as well.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 4]
```diff
@@ -1,11 +1,7 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Coding standard does not allow wildcard imports.

File: test/java/seedu/addressbook/common/UtilsTest.java [hunk 5]
```diff
@@ -43,4 +45,39 @@ private void assertAreUnique(Object... objects) {
     private void assertNotUnique(Object... objects) {
         assertFalse(Utils.elementsAreUnique(Arrays.asList(objects)));
     }
+
+    private void assertIsNull(Object... objects) {
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/common/UtilsTest.java:49]: This should not be part of this PR. In general, a PR should not contain unmerged code from other PRs. Try to limit each PR to one enhancement, which means the PR should be based on a branch that starts off from the master branch that is in sync with the upstream master branch

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @andlimey. 
Suggestions to improve further:
* update developer docs
* update jUnit tests to unit test the new functionality
* update I/O tests to test the new functionality
* limit the PR to one enhancement and avoid including changes not related to that enhancement or unmerged code from other PRs
* Pay closer attention to the coding standard
Please close the PR after viewing comments.
